lib/absinthe/relay/mutation/notation/classic.ex

```diff
@@ -100,8 +100,8 @@ defmodule Absinthe.Relay.Mutation.Notation.Classic do
       # generate both input and payload types if they are not defined within the field
       # itself. The `input` notation also autogenerates the `input` argument to the field
       quote do
-        private(:absinthe_relay, :input, {:fill, unquote(__MODULE__)})
         private(:absinthe_relay, :payload, {:fill, unquote(__MODULE__)})
+        private(:absinthe_relay, :input, {:fill, unquote(__MODULE__)})
```

Comment:
There was a discrepancy between the `modern.ex` and `classic.ex` modules in that the `absinthe_relay: list` was being generated in a different order.

For modern, it was getting generated as:
```
  absinthe_relay: [
               payload: {:fill, _},
               input: {:fill, _}
             ]
```

And classic was getting generated as:
```
  absinthe_relay: [
               input: {:fill, _},
               payload: {:fill, _} 
             ]
```

I reordered the macro in the `classic.ex` so that the list generated will be the same for both modern and classic and enable the pattern-match to work.

Reply:
Oh got it. Man that is fragile, it sort of seems like pattern matching here is the wrong call honestly. We should perhaps either switch to a map or use `if opts[:input]` or something.

Reply:
I thought of passing in some kind of param as part of the middleware call to let the middleware where it was being called from, either the mutation root or a child field, like this:

```
        middleware Absinthe.Relay.Node.ParseIDs,
         mutation_root: true,
          parent: [
            id: :parent
          ]
```

Or adding it to the mutation as a default arg along side of `input_object` that is generated